synapse/storage/data_stores/main/room.py

```diff
@@ -509,8 +506,8 @@ def _get_rooms_paginate_txn(txn):
             "get_rooms_paginate", _get_rooms_paginate_txn,
         )
 
-    @cachedInlineCallbacks(max_entries=10000)
-    def get_ratelimit_for_user(self, user_id):
+    @cached(max_entries=10000)
```

Comment:
I think it does, but have you checked `@cached` works correctly on async functions rather than those that return deferreds?

Reply:
It should work OK: it wraps things with `preserve_fn` (which wraps things in `run_in_background`, which checks for coroutines and uses `ensureDeferred` in that case).

We had some async functions cached before I had started the conversion. So I'm pretty sure it works OK!

Reply:
yup, thanks.

Interesting corollary: we can convert anything with a `cachedInlineCallbacks` wrapper to `async/await` without worrying about where it is used...

Reply:
Yeah, that's true! 👍 Could be worth doing a pass just for that since we should be able to not think too hard about it.